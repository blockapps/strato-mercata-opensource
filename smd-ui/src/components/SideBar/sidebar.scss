@import "~@blueprintjs/core/dist/blueprint.css";
@import "../../../node_modules/@blueprintjs/core/dist/variables.scss";

$navbarWidth: 260px;
aside {
  position: fixed;
  height: 100%;
  top: 0px;
  left: 0px;
  background: #394b59;
  padding: 20px;
  width: $navbarWidth;
  margin-top: 48px;
  box-shadow: inset 0 0 2px 1px rgba(16, 22, 26, 0.2), 0 1px 2px rgba(16, 22, 26, 0.4);
  z-index: 9;
}
.fa-code-branch {
  transform: rotate(180deg) scaleX(-1);
}
.menu-item{
  font-size: 1.25em;
  color: $pt-dark-text-color;
  margin-bottom: 10px;
  padding: 8px 16px;
  text-decoration: none;
  display: block;
}

.menu-text {
  margin-left: 8px;
}

.menu-item:hover {
  color: $pt-dark-code-text-color;
  text-decoration: none;
}

.active-menu-item {
  border: 1px solid $pt-dark-divider-black;
  color: $pt-dark-code-text-color !important;
  background-color: $pt-dark-code-background-color;
  text-decoration: none !important;
}

.smd-sidebar-logo {
  position: absolute;
  left: 0px;
  bottom: 60px;
  width: 100%;
  text-align: center;
}

.menu {
  min-height: 35vh;
  max-height: 60vh;
  overflow-y: auto;
}

.menu::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
  border-radius: 4px;
  background: #2F424D; 
}
 
::-webkit-scrollbar-thumb {
  border-radius: 4px; 
  background: #F4F8FA; 
}

::-webkit-scrollbar-thumb:hover {
  background: #BBCBD4; 
}

.send-tokens-dialog .chain-field {
  width: 100%;
}

#outer-container {
  position: relative;
  top: 48px;
  margin-left: $navbarWidth;
  padding: 0px 12px;
}

#menu-burger {
  display: none;
}

$collapsePoint: 800px;
$burgerSize: 35px;
@media only screen and (max-width: $collapsePoint) {
  #sidebar {
    display: none;
  }
  #outer-container {
    margin-left: 0px;
  }
  #sidebar.sidebar-expand {
    display: block;
    width: 100%;
  }
  #menu-burger {
    display: block;
    height: $burgerSize;
    width: $burgerSize;
    padding: 15px;
    cursor: pointer;
    span,
    span:before,
    span:after {
      cursor: pointer;
      border-radius: 1px;
      height: 5px;
      width: $burgerSize;
      background: $pt-dark-text-color;
      position: absolute;
      display: block;
      content: '';
      transition: all 300ms ease-in-out;
    }
    span:before {
      top: -10px;
    }
    span:after {
      bottom: -10px;
    }
    &.burger-x span {
      background-color: transparent;
      &:before,
      &:after {
        top: 0;
      }
      &:before {
        transform: rotate(45deg);
      }
      &:after {
        transform: rotate(-45deg);
      }
    }
  }
}
