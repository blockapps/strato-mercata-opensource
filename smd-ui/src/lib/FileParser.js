export function getImportStatements(source) {
    const splittedSource = source.split("\n")
    // TODO: This does not guard against multiline strings or comments
    const newImports = splittedSource.filter((line) => /^\s*import /.test(line))
    const newFileList = []
    newImports.map((item) => {
        const splittedItem = item.split(' ')
        const filename = splittedItem.filter((word) => word.includes('.sol'))
        newFileList.push(filename.length > 0 && filename[0].replace(/['";]+/g, ''))
        return item
    })
    return newFileList
}

export function replaceImportStatementsWithSource(source, importsCode, tabData) {
    const splittedSource = source.split("\n")
    const filteredList = splittedSource.map((item) => {
        const code = importsCode.filter(
            (codeItem) =>
                item.includes(codeItem.title)
        )
        return code.length > 0 ? getFileAndReplaceWithImport(code[0].text, tabData) : item }
    )
    filteredList.join("\r\n")
    return filteredList
}

export function getFileAndReplaceWithImport(code, tabData) {
    const newFileList = getImportStatements(code)
    if (newFileList.length === 0) {
        return code
    }
    const items = tabData.filter((item) =>
        newFileList.includes(item.title)
    )
    const replacedCode = replaceImportStatementsWithSource(code, items, tabData)
    return replacedCode.join("\r\n")
}
