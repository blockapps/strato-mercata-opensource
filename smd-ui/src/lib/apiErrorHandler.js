export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
