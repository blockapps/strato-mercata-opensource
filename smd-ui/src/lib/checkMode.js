import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}