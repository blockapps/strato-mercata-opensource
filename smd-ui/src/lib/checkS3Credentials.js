import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
