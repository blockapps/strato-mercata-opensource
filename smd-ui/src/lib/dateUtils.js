import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
