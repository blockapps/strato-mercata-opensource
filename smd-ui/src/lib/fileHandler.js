export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}