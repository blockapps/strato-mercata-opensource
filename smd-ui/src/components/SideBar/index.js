import React, { Component } from 'react';
import './sidebar.css';
import { NavLink } from 'react-router-dom';
import logo from './BlockAppsLogos_DarkBG-Horizontal.png';

import 'normalize.css/normalize.css';
import '@blueprintjs/core/dist/blueprint.css';
import mixpanelWrapper from '../../lib/mixpanelWrapper';
import { isS3Available } from '../../lib/checkS3Credentials';

class SideBar extends Component {

  // noOverlay
  // TODO: customCrossIcon={<div><div className="pt-icon-standard pt-icon-chevron-left"/></div>}
  render() {
    const navLinksData = (
      [
        //{path: '/nodes', label: 'Nodes', id: 'nodes', icon: "pt-icon-layout-auto"},
        { path: '/home', label: 'Network Stats', id: 'network_stats', icon: "fa-chart-simple" },
        { path: '/accounts', label: 'Users', id: 'accounts', icon: "fa-users" },
        { path: '/transactions', label: 'Transactions', id: 'transactions', icon: "fa-exchange" },
        { path: '/contracts', label: 'Contracts', id: 'contracts', icon: "fa-file-contract" },
        { path: '/blocks', label: 'Blocks', id: 'blocks', icon: "fa-cube" },
        { path: '/code_editor', label: 'Contract Editor', id: 'code_editor', icon: "fa-code" },
      ]
    );

    return (
      <aside id="sidebar" className={this.props.isCollapsed ? '' : 'sidebar-expand'}>
        <div className="menu">
          {
            navLinksData.map(data =>
              <NavLink
                key={data.id}
                id={data.id}
                to={data.path}
                className="menu-item"
                activeClassName="active-menu-item"
                onClick={this.props.toggleCollapse}
              >
                <i className={'fa ' + data.icon}> </i>
                <span className="menu-text"> {data.label}</span>
              </NavLink>
            )
          }
          {isS3Available() &&
            <NavLink
              id={'external_storage'}
              to={'/external_storage'}
              className="menu-item"
              activeClassName="active-menu-item"
              onClick={() => { mixpanelWrapper.track('nav_link_external_storage_click') }}
            >
              <i className='fa fa-cloud-upload'> </i>
              <span className="menu-text">External Storage</span>
            </NavLink>}
        </div>
        <div className="smd-sidebar-logo">
          <a href="http://blockapps.net" target="_blank" rel="noopener noreferrer">
            <img
              src={logo}
              width="120"
              alt="Blockapps Logo"
            />
          </a>
        </div>
      </aside>
    );
  }
}

export default SideBar;
