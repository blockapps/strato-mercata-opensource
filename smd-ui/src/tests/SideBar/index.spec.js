import React from 'react';
import SideBar from "../../components/SideBar";
import * as checkMode from '../../lib/checkMode';
import * as checkS3Credentials from '../../lib/checkS3Credentials';

describe('SideBar: index', () => {
  let wrapper;

  test('render component for public mode', () => {
    checkMode.isOauthEnabled = jest.fn().mockReturnValue(true);
    checkS3Credentials.isS3Available = jest.fn().mockReturnValue(false);
    wrapper = shallow(<SideBar />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('render component for non oauth mode', () => {
    checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    checkS3Credentials.isS3Available = jest.fn().mockReturnValue(true);
    wrapper = shallow(<SideBar />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('sequence of links for oauth and non-oauth mode', () => {

    beforeEach(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
      checkS3Credentials.isS3Available = jest.fn().mockReturnValue(true);
      wrapper = shallow(<SideBar />);
    });

    test('first position /home', () => {
      wrapper.find('NavLink').at(0).simulate('click');
      expect(wrapper.find('NavLink').get(0)).toMatchSnapshot();
    });

    test('third position /blocks', () => {
      wrapper.find('NavLink').at(2).simulate('click');
      expect(wrapper.find('NavLink').get(2)).toMatchSnapshot();
    });

    test('fourth position /transactions', () => {
      wrapper.find('NavLink').at(3).simulate('click');
      expect(wrapper.find('NavLink').get(3)).toMatchSnapshot();
    });

    test('fifth position /accounts', () => {
      wrapper.find('NavLink').at(4).simulate('click');
      expect(wrapper.find('NavLink').get(4)).toMatchSnapshot();
    });

    test('sixth position /contracts', () => {
      wrapper.find('NavLink').at(5).simulate('click');
      expect(wrapper.find('NavLink').get(5)).toMatchSnapshot();
    });

    test('seventh position /code_editor', () => {
      wrapper.find('NavLink').at(6).simulate('click');
      expect(wrapper.find('NavLink').get(6)).toMatchSnapshot();
    });
    
    test('eighth position /external-storage', () => {
      wrapper.find('NavLink').at(1).simulate('click');
      expect(wrapper.find('NavLink').get(1)).toMatchSnapshot();
    });

  });

});