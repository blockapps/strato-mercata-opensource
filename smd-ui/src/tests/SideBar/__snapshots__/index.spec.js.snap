// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar: index render component for non oauth mode 1`] = `
"<aside id=\\"sidebar\\" className=\\"sidebar-expand\\">
  <div className=\\"menu\\">
    <NavLink id=\\"network_stats\\" to=\\"/home\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-chart-simple\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Network Stats
      </span>
    </NavLink>
    <NavLink id=\\"accounts\\" to=\\"/accounts\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-users\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Users
      </span>
    </NavLink>
    <NavLink id=\\"transactions\\" to=\\"/transactions\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-exchange\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Transactions
      </span>
    </NavLink>
    <NavLink id=\\"contracts\\" to=\\"/contracts\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-file-contract\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Contracts
      </span>
    </NavLink>
    <NavLink id=\\"blocks\\" to=\\"/blocks\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-cube\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Blocks
      </span>
    </NavLink>
    <NavLink id=\\"code_editor\\" to=\\"/code_editor\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-code\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Contract Editor
      </span>
    </NavLink>
    <NavLink id=\\"external_storage\\" to=\\"/external_storage\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[Function: onClick]}>
      <i className=\\"fa fa-cloud-upload\\">
         
      </i>
      <span className=\\"menu-text\\">
        External Storage
      </span>
    </NavLink>
  </div>
  <div className=\\"smd-sidebar-logo\\">
    <a href=\\"http://blockapps.net\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
      <img src={{...}} width=\\"120\\" alt=\\"Blockapps Logo\\" />
    </a>
  </div>
</aside>"
`;

exports[`SideBar: index render component for public mode 1`] = `
"<aside id=\\"sidebar\\" className=\\"sidebar-expand\\">
  <div className=\\"menu\\">
    <NavLink id=\\"network_stats\\" to=\\"/home\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-chart-simple\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Network Stats
      </span>
    </NavLink>
    <NavLink id=\\"accounts\\" to=\\"/accounts\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-users\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Users
      </span>
    </NavLink>
    <NavLink id=\\"transactions\\" to=\\"/transactions\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-exchange\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Transactions
      </span>
    </NavLink>
    <NavLink id=\\"contracts\\" to=\\"/contracts\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-file-contract\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Contracts
      </span>
    </NavLink>
    <NavLink id=\\"blocks\\" to=\\"/blocks\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-cube\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Blocks
      </span>
    </NavLink>
    <NavLink id=\\"code_editor\\" to=\\"/code_editor\\" className=\\"menu-item\\" activeClassName=\\"active-menu-item\\" onClick={[undefined]}>
      <i className=\\"fa fa-code\\">
         
      </i>
      <span className=\\"menu-text\\">
         
        Contract Editor
      </span>
    </NavLink>
  </div>
  <div className=\\"smd-sidebar-logo\\">
    <a href=\\"http://blockapps.net\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
      <img src={{...}} width=\\"120\\" alt=\\"Blockapps Logo\\" />
    </a>
  </div>
</aside>"
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode eighth position /external-storage 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="accounts"
  onClick={undefined}
  to="/accounts"
>
  <i
    className="fa fa-users"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Users
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode fifth position /accounts 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="blocks"
  onClick={undefined}
  to="/blocks"
>
  <i
    className="fa fa-cube"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Blocks
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode first position /home 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="network_stats"
  onClick={undefined}
  to="/home"
>
  <i
    className="fa fa-chart-simple"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Network Stats
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode fourth position /transactions 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="contracts"
  onClick={undefined}
  to="/contracts"
>
  <i
    className="fa fa-file-contract"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Contracts
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode seventh position /code_editor 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="external_storage"
  onClick={[Function]}
  to="/external_storage"
>
  <i
    className="fa fa-cloud-upload"
  >
     
  </i>
  <span
    className="menu-text"
  >
    External Storage
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode sixth position /contracts 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="code_editor"
  onClick={undefined}
  to="/code_editor"
>
  <i
    className="fa fa-code"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Contract Editor
  </span>
</NavLink>
`;

exports[`SideBar: index sequence of links for oauth and non-oauth mode third position /blocks 1`] = `
<NavLink
  activeClassName="active-menu-item"
  className="menu-item"
  id="transactions"
  onClick={undefined}
  to="/transactions"
>
  <i
    className="fa fa-exchange"
  >
     
  </i>
  <span
    className="menu-text"
  >
     
    Transactions
  </span>
</NavLink>
`;
